applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c

```diff
@@ -683,6 +707,25 @@ int le_audio_send(uint8_t const *const data, size_t size)
 		return 0;
 	}
 
+	/* net_buf_alloc allocates buffers for APP->NET transfer over HCI RPMsg,
+	 * but when these buffers are released it is not guaranteed that the
+	 * data has actually been sent. The data might be queued on the NET core,
+	 * and this can cause delays in the audio.
+	 * When stream_sent_cb() is called the data has been sent.
+	 * Data will be discarded if allocation becomes too high, to avoid audio delays.
+	 * If the NET and APP core operates in clock sync, discarding should not occur.
+	 */
+	if (atomic_get(&iso_tx_pool_alloc) >= HCI_ISO_BUF_ALLOC_PER_CHAN) {
```

Comment:
I would appreciate a walkthrough of this PR. 
Should this be done in some of the other le_audio_*** files as well?